### ISSUE TITLE
Indistinguishable dynamic profiles

### ISSUE THREAD

**Issue by oughnic** [MauroDataMapper/mdm-ui#402]
**Describe the bug**
When adding a profile that has branches and versions to a data model multiple indistinguishable profiles are listed in the drop down

**To Reproduce**
1. Edit a profile with branching
2. Save
3. Try adding that profile to an existing model

**Expected behavior**
Only the current finalised profile is offered
**Screenshots**
![image](https://user-images.githubusercontent.com/27196895/144074932-ca9ff6af-ccab-483a-b783-cd371bf92438.png)

**Comment by olliefreeman** [comment]
the UI is getting back profiles with different versions you really need to show the version in the name profiles are recognised by the NS, name & version and here we have multiple models with the same name but different versions so arguably the core has done all it can.


having said all that ... theres another issue where the dynamic profiles have been setup to use the "tag" or branchname as their version ... which breaks the version sorting
i want to change this so it uses the modelVersion only ... but i havent looked at it yet...

but the UI will def need to show the version in the list naming..but better wait for the other issue fix first

**Comment by olliefreeman** [comment]
Wait on [mdm-core#209](https://github.com/MauroDataMapper/mdm-core/issues/209)
